python/pyarrow/tests/test_csv.py

```diff
@@ -654,6 +654,19 @@ def row_num(x):
         expected_rows = [InvalidRow(2, 1, row_num(2), "c")]
         assert parse_opts.invalid_row_handler.rows == expected_rows
 
+        # Test ser/de
+        parse_opts.invalid_row_handler = InvalidRowHandler('skip')
+        state = parse_opts.__getstate__()
+
+        parse_opts = ParseOptions()
+        parse_opts.__setstate__(state)
```

Comment:
Instead of calling getstate/setstate explicitly, can you pickle/unpickle instead? (like `new_parse_opts = pickle.loads(pickle.dumps(parse_opts))`)

Reply:
Sure, updated!

I've also updated `__getstate__` to return `self.invalid_row_handler` instead of `self._invalid_row_handler`, let me know if this looks good to you.